### PR TITLE
affinity, memory limit, healthcheck library, refactoring

### PR DESCRIPTION
**Pull request by SCBbestof**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -48,13 +57,34 @@ jobs:
...
+            source ${BASH_ENV}
+            docker build \
+            --build-arg SONATYPE_USER=${SONATYPE_USER} \
+            --build-arg SONATYPE_PASSWORD=${SONATYPE_PASSWORD} \
+            --tag nexus.in.ft.com:5000/coco/${CIRCLE_PROJECT_REPONAME}:${DOCKER_TAG} .
+      - deploy:
+          name: Push Docker image
+          command: |
+            source ${BASH_ENV}
+            docker login nexus.in.ft.com:5000 -u ${DOCKER_USER} -p ${DOCKER_PASSWORD}
```
Comment (peteclark-ft) [inline .circleci/config.yml]: We currently build the docker image in dockerhub, any reason we're now pushing to Nexus? This is also the first time I've seen us push images from a CircleCI build - any reason for this?

File: resources/healthchecks.go [hunk 2]
```diff
@@ -9,38 +9,72 @@ import (
...
 	"github.com/Financial-Times/publish-carousel/cms"
 	"github.com/Financial-Times/publish-carousel/native"
 	"github.com/Financial-Times/publish-carousel/s3"
 	"github.com/Financial-Times/publish-carousel/scheduler"
 	log "github.com/Sirupsen/logrus"
+	"github.com/Financial-Times/service-status-go/gtg"
 )
 
+type HealthService struct {
+	config *config
```
Comment (peteclark-ft) [inline resources/healthchecks.go]: I think typically (as per the cookiecutter) we embed the `v1_1.HealthCheck` struct in the `HealthService`, this automatically pulls in the `systemCode`, `appName` and `description` so you don't need a new `config` struct.

File: resources/healthchecks.go [hunk 3]
```diff
@@ -9,38 +9,72 @@ import (
...
+	}
+	service.Checks = service.getHealthchecks(db, s3Service, notifier, sched, configError, upServices...)
+	return service
+}
+
 // Health returns a handler for the standard FT healthchecks
-func Health(db native.DB, s3Service s3.ReadWriter, notifier cms.Notifier, sched scheduler.Scheduler, configError error, upServices ...cluster.Service) func(w http.ResponseWriter, r *http.Request) {
-	return fthealth.HandlerParallel("publish-carousel", "A microservice that continuously republishes content and annotations available in the native store.", getHealthchecks(db, s3Service, notifier, sched, configError, upServices...)...)
+func (healthService *HealthService) Health() func(w http.ResponseWriter, r *http.Request) {
+	hc := fthealth.HealthCheck{
```
Comment (peteclark-ft) [inline resources/healthchecks.go]: I think we need this to be a `TimedHealthCheck` now with a 10s timeout. Also if the `HealthService` has an embedded `HealthCheck` type, you can then do:

```
health.TimedHealthCheck{
   HealthCheck: healthService.HealthCheck,
   Timeout: 10 * time.Second,
}
```

File: .circleci/config.yml (hunk 1 above)
Comment (SCBbestof) [reply .circleci/config.yml]: Removed.

**Pull request merged**